### PR TITLE
Split CoreNode interface

### PR DESCRIPTION
**Pull request by ianopolous**
Ignore the branch name, this is a precursor to moving the corenode PKI stuff into IPFS.

This PR removes the follow request calls from CoreNode into SocialNetwork, and changes the http path of the social calls.

### CODE REVIEW

File: src/peergos/server/social/HttpSocialNetworkServer.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    private final HttpServer server;
+    private final InetSocketAddress address;
+    private final SocialHandler ch;
+
+    public HttpSocialNetworkServer(SocialNetwork social, MutablePointers mutable, InetSocketAddress address) throws IOException
+    {
+
+        this.address = address;
+        if (address.getHostName().contains("local"))
```
Comment (kevodwyer) [inline src/peergos/server/social/HttpSocialNetworkServer.java]: case insensitive?

File: src/peergos/server/social/HttpSocialNetworkServer.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    exchange.getResponseHeaders().set("Trailer", cause.getMessage());
+                else
+                    exchange.getResponseHeaders().set("Trailer", e.getMessage());
+
+                exchange.sendResponseHeaders(400, 0);
+            } finally {
+                exchange.close();
+                long t2 = System.currentTimeMillis();
+                if (LOGGING)
+                    System.out.println("Corenode server handled " + method + " request in: " + (t2 - t1) + " mS");
```
Comment (kevodwyer) [inline src/peergos/server/social/HttpSocialNetworkServer.java]: msg mentions corenode

File: src/peergos/server/social/HttpSocialNetworkServer.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        try {
+            String hostname = args.getArg("domain", "localhost");
+            System.out.println("Starting social network server listening on: " + hostname+":"+port +" proxying to "+social);
+            InetSocketAddress address = new InetSocketAddress(hostname, port);
+            HttpSocialNetworkServer server = new HttpSocialNetworkServer(social, mutable, address);
+            server.start();
+        } catch (Exception e)
+        {
+            e.printStackTrace();
+            System.out.println("Couldn't start Corenode server!");
```
Comment (kevodwyer) [inline src/peergos/server/social/HttpSocialNetworkServer.java]: corenode ref again

File: src/peergos/server/social/HttpSocialNetworkServer.java (hunk 1 above)
Comment (ianopolous) [reply src/peergos/server/social/HttpSocialNetworkServer.java]: InetSocketAddress already does that for us

File: src/peergos/server/social/HttpSocialNetworkServer.java (hunk 2 above)
Comment (ianopolous) [reply src/peergos/server/social/HttpSocialNetworkServer.java]: done

File: src/peergos/server/social/HttpSocialNetworkServer.java (hunk 3 above)
Comment (ianopolous) [reply src/peergos/server/social/HttpSocialNetworkServer.java]: fixed

File: src/peergos/server/social/HttpSocialNetworkServer.java (hunk 1 above)
Comment (ianopolous) [reply src/peergos/server/social/HttpSocialNetworkServer.java]: fixed

**Pull request merged**